cql/DashboardLibrary.cql

```diff
@@ -160,40 +160,49 @@ context Patient
         // low to avoid competing with existing cytology results
-        when c ~ "AGC Favor Neoplasia" then {text: 'AGC-FN', rank: 1}
-        when c ~ "Atypical Endocervical Cells" then {text: 'Atypical Endocervical Cells', rank: 1}
-        when c ~ "Atypical Glandular Cells Uncertain" then {text: 'AGC-US', rank: 1}
-        when c in "AIS" then {text: 'AIS', rank: 1}
-        when c ~ "Endocervical AIS" then {text: 'Endocervical AIS', rank: 1}
-        when c ~ "Endocervical Cells Favor Neoplasia" then {text: 'Endocervical Cells Favor Neoplasia', rank: 1}        
-        when c ~ "Atypical Endometrial Cells" then {text: 'Atypical Endometrial Cells', rank: 1}
-
-        when c ~ "Histiocytes" then {text: 'Histiocytes', rank: 1}
-        when c ~ "Benign Endometrial Cells" then {text: 'Benign Endometrial Cells', rank: 1}
-        when c ~ "Benign Glandular Cells" then {text: 'Benign Glandular Cells', rank: 1}
-        when c ~ "Endometrial Stromal Cells" then {text: 'Endometrial stromal cells', rank: 1}
-        when c ~ "Unsatisfactory" then {text: 'Unsatisfactory', rank: 1}
-        when c ~ "Absent Transformation Zone" then {text: 'Absent Transformation Zone', rank: 1}
-
-        when c is null then {text: 'NULL', rank: -1}
-        else {text: 'UNK', rank: 1}
+        when c ~ "AGC Favor Neoplasia" then {riskTableInput: 'AGC-FN', label: 'AGC-FN', rank: 1, grade: 2}
+        when c ~ "Atypical Endocervical Cells" then {riskTableInput: 'AECC', label: 'Atypical Endocervical Cells', rank: 1, grade: 2}
+        when c ~ "Atypical Glandular Cells Uncertain" then {riskTableInput: 'AGC-US', label: 'AGC-US', rank: 1, grade: 2}
+        when c in "AIS" then {riskTableInput: 'AIS', label: 'AIS', rank: 1, grade: 2}
+        when c ~ "Endocervical AIS" then {riskTableInput: 'ECAIS', label: 'Endocervical AIS', rank: 1, grade: 2}
+        when c ~ "Endocervical Cells Favor Neoplasia" then {riskTableInput: 'ECFN', label: 'Endocervical Cells Favor Neoplasia', rank: 1, grade: 2}
+        when c ~ "Atypical Endometrial Cells" then {riskTableInput: 'AEMC', label: 'Atypical Endometrial Cells', rank: 1, grade: 1}
+
+        when c ~ "Histiocytes" then {riskTableInput: 'HIST', label: 'Histiocytes', rank: 1, grade: 1}
+        when c ~ "Benign Endometrial Cells" then {riskTableInput: 'BEMC', label: 'Benign Endometrial Cells', rank: 1, grade: 1}
+        when c ~ "Benign Glandular Cells" then {riskTableInput: 'BGC', label: 'Benign Glandular Cells', rank: 1, grade: 1}
+        when c ~ "Endometrial Stromal Cells" then {riskTableInput: 'EMSC', label: 'Endometrial stromal cells', rank: 1, grade: 1}
+        when c ~ "Unsatisfactory" then {riskTableInput: 'UNSAT', label: 'Unsatisfactory', rank: 1, grade: 1}
+        when c ~ "Absent Transformation Zone" then {riskTableInput: 'ATZ', label: 'Absent Transformation Zone', rank: 1, grade: 1}
+
+        when c is null then {riskTableInput: 'NULL', label: 'None', rank: -1, grade: 0}
```

Comment:
All three interpretation functions now say 'No result found' for the null case.

Reply:
Nice! I added a comment https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/153#discussion_r1882851516 about changing the `else` condition to also show "No result found" instead of "None", so the two are consistent.